pkg/handlers/v1/ec2.go

```diff
@@ -69,40 +78,113 @@ type ec2ChangedPropsARN struct {
 
 type ec2Transformer struct{}
 
-func (t ec2Transformer) Create(event awsConfigEvent) (Output, error) {
-	output, err := getBaseOutput(event.ConfigurationItem)
+func (t ec2Transformer) Create(event awsConfigEvent) ([]Output, error) {
+	outputs := []Output{}
+	baseOutput, err := getBaseOutput(event.ConfigurationItem)
 	if err != nil {
-		return Output{}, err
+		return []Output{}, err
 	}
-
 	// if a resource is created for the first time, there is no diff.
 	// just read the configuration
 	var config ec2Configuration
 	if err := json.Unmarshal(event.ConfigurationItem.Configuration, &config); err != nil {
-		return Output{}, err
+		return []Output{}, err
 	}
-	change := extractEC2NetworkInfo(&config)
-	change.ChangeType = added
-	output.Changes = append(output.Changes, change)
-	return output, nil
+
+	hasEni := false
+	for _, ni := range config.NetworkInterfaces {
+		ni := ni
+		hasEni = true
+		output := baseOutput
+		change := Change{}
+		output.ChangeTime = ni.Attachment.AttachTime
+		private, public, dns := extractNetworkInterfaceInfo(&ni)
+		change.PrivateIPAddresses = append(change.PrivateIPAddresses, private...)
+		change.PublicIPAddresses = append(change.PublicIPAddresses, public...)
+		change.Hostnames = append(change.Hostnames, dns...)
+		change.ChangeType = added
+		output.Changes = append(output.Changes, change)
+		outputs = append(outputs, output)
+	}
+	if !hasEni {
+		return []Output{}, nil // Reject EC2s with no ENI
+	}
+
+	return outputs, nil
 }
 
-func (t ec2Transformer) Update(event awsConfigEvent) (Output, error) {
-	output, err := getBaseOutput(event.ConfigurationItem)
+func initializeAttachMapEntry(attachMap map[string]map[string][]string, attachTime string) {
+	attachMap[attachTime] = make(map[string][]string)
+	attachMap[attachTime][private] = make([]string, 0)
+	attachMap[attachTime][public] = make([]string, 0)
+	attachMap[attachTime][dns] = make([]string, 0)
+}
+
+func separateAddedChanges(baseOutput Output, addedChange Change, privateIPMap map[string]string, publicIPMap map[string]string, hostnameMap map[string]string) []Output {
+	outputs := make([]Output, 0)
+	// attachMap is a map of attachTimes (string) to a map with the following pairs:
+	// 	"privateIPs": array of private IPs
+	// 	"publicIPs": array of public IPs
+	// 	"hostnames": array of hostnames
+	attachMap := make(map[string]map[string][]string)
```

Comment:
I think if you're at a point where you need a complex comment like this then it may be beneficial to create a struct instead. I think nested maps is a bit hard to follow.

Reply:
- [ ] Convert map-of-maps into dedicated struct